library/core/src/main/java/com/google/android/exoplayer2/scheduler/Requirements.java

```diff
@@ -162,6 +175,34 @@ private boolean isDeviceIdle(Context context) {
         : Util.SDK_INT >= 20 ? !powerManager.isInteractive() : !powerManager.isScreenOn();
   }
 
+  /**
+   * Implementation taken from the the WorkManager source.
+   * @see <a href="https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/work/workmanager/src/main/java/androidx/work/impl/constraints/trackers/StorageNotLowTracker.java">StorageNotLowTracker</a>
+   */
+  private boolean isStorageNotLow(Context context) {
+    IntentFilter intentFilter = new IntentFilter();
+    intentFilter.addAction(Intent.ACTION_DEVICE_STORAGE_OK);
+    intentFilter.addAction(Intent.ACTION_DEVICE_STORAGE_LOW);
+    Intent intent = context.registerReceiver(null, intentFilter);
+    if (intent == null || intent.getAction() == null) {
+      // ACTION_DEVICE_STORAGE_LOW is a sticky broadcast that is removed when sufficient
+      // storage is available again.  ACTION_DEVICE_STORAGE_OK is not sticky.  So if we
+      // don't receive anything here, we can assume that the storage state is okay.
+      return true;
+    } else {
+      switch (intent.getAction()) {
+        case Intent.ACTION_DEVICE_STORAGE_OK:
+          return true;
```

Comment:
I guess based on the comment above that you copied this code from elsewhere, but if `ACTION_DEVICE_STORAGE_OK` is not sticky, how does this case ever occur :)? Seems to me we can completely remove `ACTION_DEVICE_STORAGE_OK` from this case without any change in behavior?

Reply:
As a summary, I wonder whether this entire method can just be:
```
@Nullable Intent intent =
    context.registerReceiver(null, new IntentFilter(Intent.ACTION_DEVICE_STORAGE_LOW));
return intent == null;
```

Reply:
No need to fix this. I can do it when I merge. Although please check whether the proposed alternative is sufficient!

Reply:
You are correct! Tested